pytensor/link/pytorch/dispatch/scalar.py

```diff
@@ -38,3 +38,34 @@ def pytorch_func(*args):
             )
 
     return pytorch_func
+
+
+@pytorch_funcify.register(ScalarLoop)
+def pytorch_funicify_ScalarLoop(op, node, **kwargs):
+    update = pytorch_funcify(op.fgraph)
+    state_length = op.nout
+    if op.is_while:
+
+        def scalar_loop(steps, *start_and_constants):
+            carry, constants = (
+                start_and_constants[:state_length],
+                start_and_constants[state_length:],
+            )
+            done = True
+            for _ in range(steps):
+                *carry, done = update(*carry, *constants)
+                if done:
+                    break
+            return torch.stack(carry), torch.tensor([done])
+    else:
+
+        def scalar_loop(steps, *start_and_constants):
+            carry, constants = (
+                start_and_constants[:state_length],
+                start_and_constants[state_length:],
+            )
+            for _ in range(steps):
+                carry = update(*carry, *constants)
+            return torch.stack(carry)
```

Comment:
Why is the stack needed?

Reply:
This could be another byproduct of how the test is written; but if you remove the stack you get

```

    def output_filter(self, var: Variable, out: Any) -> Any:
>       return out.cpu()
E       AttributeError: 'list' object has no attribute 'cpu'
E       Apply node that caused the error: ScalarLoop(n_steps, x0)
E       Toposort index: 0
E       Inputs types: [ScalarType(int64), ScalarType(float64)]
E       Inputs shapes: [(), ()]
E       Inputs strides: [(), ()]
E       Inputs values: [20, 0.0]
E       Outputs clients: [[output[0](ScalarLoop.0)], [output[1](ScalarLoop.1)]]

FAILED tests/link/pytorch/test_basic.py::test_ScalarLoop - AttributeError: 'tuple' object has no attribute 'cpu'
```

Reply:
fixed